### ISSUE TITLE
Develop web UI to enable stacking of the citycoin

### ISSUE THREAD

**Issue by pxydn** [syvita/citycoin#10]
- [ ] a Stacker can submit citycoin to the contract using the Stacks Web Wallet
- [ ] a Stacker can choose the number of reward cycles to participate in
- [ ] a Stacker can see available STX rewards and claim them
- [ ] a Stacker can see their Stacking history, stored in Gaia (TBF)

**Comment by friedger** [comment]
Is a clone of https://stacks-send-many.pages.dev/ good enough for a first UI?

**Comment by 314159265359879** [comment]
> 
> 
> Is a clone of https://stacks-send-many.pages.dev/ good enough for a first UI?

Excellent idea, making it prettier could be a nice to have? since we are on a clock I don't think we have to be picky for MVP

**Comment by pxydn** [comment]
> > 
> 
> > 
> 
> > Is a clone of https://stacks-send-many.pages.dev/ good enough for a first UI?
> 
> 
> 
> Excellent idea, making it prettier could be a nice to have? since we are on a clock I don't think we have to be picky for MVP

we do have devs that could throw something together using some of the code from this using the new design systems. cc @syvita/ui

**Comment by whoabuddy** [comment]
Yes, and yes! @pxydn that page is here on GH: https://github.com/friedger/stacks-send-many

**Comment by pxydn** [comment]
> Yes, and yes! @pxydn that page is here on GH: https://github.com/friedger/stacks-send-many

yes sir! will look at it 💪

**Comment by 314159265359879** [comment]
Would the stacking/mining UI also need a dashboard such as stacking.club/stxmining.club provide for Stacks?
This system is simpler, because there is no interruption with prep fases but some details could be helpful.  

As a citycoin miner I would like to see how many miners there are and perhaps what the average commitment is. 
As a citycoin stacker I would like to see the payout in citycoin per committed stx (not ustx) of the last _x_ blocks.

**Comment by pxydn** [comment]
> Would the stacking/mining UI also need a dashboard such as stacking.club/stxmining.club provide for Stacks?
> This system is simpler, because there is no interruption with prep fases but some details could be helpful.
> 
> As a citycoin miner I would like to see how many miners there are and perhaps what the average commitment is.
> As a citycoin stacker I would like to see the payout in citycoin per committed stx (not ustx) of the last _x_ blocks.

Yeah, it should be a mix of all of these with all their features.

**Comment by whoabuddy** [comment]
Yes, we definitely want to provide stats, and that will be a big goal of adding read-only functions to the contract. I want to expose as much data as possible and we can choose to consume and display the easiest first in the UI.

**Comment by friedger** [comment]
Should the UI go into a separate repo, e.g. `citycoin-web`?

**Comment by pxydn** [comment]
> Should the UI go into a separate repo, e.g. `citycoin-web`?

probs a good idea

**Comment by whoabuddy** [comment]
@friedger Yes please, I created one for citystacks.co that would be the main marketing site, including the brand, a map of who we're targeting, who's coming next, who picked up the bag, etc, and stuff like that. I like `citystacks-tools`.

The UI for mining and Stacking could live on a subdomain, and a separate repo would be easiest for publishing, we can use the Cloudflare Wrangler GH action: https://github.com/cloudflare/wrangler-action (happy to set that up, too)

**Comment by pxydn** [comment]
> @friedger Yes please, I created one for citystacks.co that would be the main marketing site, including the brand, a map of who we're targeting, who's coming next, who picked up the bag, etc, and stuff like that. I like `citystacks-tools`.
> 
> The UI for mining and Stacking could live on a subdomain, and a separate repo would be easiest for publishing, we can use the Cloudflare Wrangler GH action: https://github.com/cloudflare/wrangler-action (happy to set that up, too)

we'll use CF Pages, not workers for this. have you bought the domain, also? I thought we were branding everything as Citycoins? please update me on this.

for the rest, yes, it's probably a good idea.
